### PR TITLE
First reconciliation PR from production/RRFS.v1

### PR DESCRIPTION
**Pull request by grantfirl**
Contains the original PR from @haiqinli for the production/RRFS.v1 branch:

https://github.com/ufs-community/ccpp-physics/pull/176

on top of the latest ufs/dev branch.

Note that the changes to drag_suite.F90 have been superceded by changes already in ufs/dev.

### CODE REVIEW

File: physics/CONV/Grell_Freitas/cu_gf_deep.F90 [hunk 1]
```diff
@@ -4118,7 +4119,7 @@ subroutine cup_output_ens_3d(xff_mid,xf_ens,ierr,dellat,dellaq,dellaqc,  &
         ,intent (inout)                   ::                           &
         ierr,ierr2,ierr3
      integer, intent(in) :: dicycle
-     real(kind=kind_phys),    intent(in), dimension (its:ite) :: xf_dicycle
+     real(kind=kind_phys),    intent(in), dimension (its:ite) :: xf_dicycle, rrfs_factor
 !$acc declare copyin(zu,pwd,p_cup,sig,xmbm_in,xmbs_in,edt,xff_mid,dellat,dellaqc,dellaq,pw,ktop,xland1,xf_dicycle)
```
Comment (dustinswales) [inline physics/CONV/Grell_Freitas/cu_gf_deep.F90:4123]: Maybe irrelevant, but if xf_dicycle is part to the $acc directive, should rrfs_factor be added?

Comment (grantfirl) [reply physics/CONV/Grell_Freitas/cu_gf_deep.F90:4123]: @dustinswales If this caused a problem, do you think that this type of thing would be caught in the SCM nvhpc CI test? Just curious. I.e. is this something that the compiler would catch or would it just be a run-time issue (failure or performance hit)?

Comment (dustinswales) [reply physics/CONV/Grell_Freitas/cu_gf_deep.F90:4123]: I have no clue if this would get caught at buildtime/runtime.

**Review by tanyasmirnova — APPROVED** [review]
I approve the changes in RUC LSM.

**Review by rhaesung — APPROVED** [review]

**Comment by jkbk2004** [comment]
Tests are done at https://github.com/ufs-community/ufs-weather-model/pull/2449. @rhaesung @dustinswales can you merge this pr?

**Pull request merged**